src/test/java/io/bf2/kafka/authorizer/CustomAclAuthorizerTest.java

```diff
@@ -245,6 +250,29 @@ void testCreateAclsDeniedForInvalidBinding() throws IOException {
         createAclsAndExpect("owner1", bindings, (aclCreateResult) -> isFailedWithError(aclCreateResult, CustomAclAuthorizer.CREATE_ACL_INVALID_BINDING), 3);
     }
 
+    @Test
+    void testOrphanedPrincipalsDeletedFromDelegateOnStart() throws Exception {
+        PartitionCounter partitionCounter = generateMockPartitionCounter(1001, false, false);
+        try (CustomAclAuthorizer auth = new CustomAclAuthorizer(this.delegate, partitionCounter)) {
+            auth.configure(config);
+            List<AclBindingFilter> expected = new ArrayList<>();
```

Comment:
Can we move `expected` out of the try block so its clearer its expected state and not part of the method under test?

I presume the expected list is modified from the default list in some way but I'm not clear what is different and why that tests the behaviour mentioned in the test name.

Reply:
Most of the tests are configured by a single properties file, this is asserting that we tell the delegate to delete all rules for principals with rules in that propsfile. 

Maybe in this case it would be better to construct the input properties so it can be more like, `given principal X has a custom acl rule; when we configure the plugin, we try to delete rules for X`.

Reply:
I changed it to use configuration controlled by the test case rather than the default propsfile, and updated the name to be more in line with the behaviour